avm-transpiler/src/transpile.rs

```diff
@@ -992,7 +992,7 @@ fn handle_storage_read(
     inputs: &Vec<ValueOrArray>,
 ) {
     // For the foreign calls we want to handle, we do not want inputs, as they are getters
-    assert!(inputs.len() == 1); // storage_slot
+    assert!(inputs.len() == 2); // output, len - but we dont use this len - its for the oracle
```

Comment:
We need a bit more detail here. In particular explain that the destination already has a length, and that is the one that will be used. (and why do we need this extra length? what does "the oracle" mean)

However... ideally we'd have an assert here that `length==destination.length`, if you can force the length to be known at compile time.

If you can't I guess that's life.

Reply:
Clarified a bit in the comment (first just reverted the change), but I don't see an easy way to make the second assertion, since the destination size is known at this point, but the second argument is just a memory address. Is it possible to "unwrap" it somehow?

Reply:
Length should be known at compile time btw, since the output of the opcode is an array explicitly (not a slice)

Reply:
The only way for the transpiler to do it is to emit a const opcode with destination.length and an opcode that checks equality between that one and the second argument. But without scratch space it's a pain. (should I reserve scratch space at this point? it has come up several times)

Reply:
If this is part of a larger discussion, I would leave as it is for the time being and try to get this in, since it's blocking TXE usage in public ATM (and this was already there, just removed recently)

Reply:
Yeah, I'm ok with it as it is. The question for Alvaro was more on the Noir/Brillig side. If there was any way to mark length as `comptime`/etc so that it gets to the transpiler as a resolved number.

Reply:
That'd be possible but I think there is no way to have a literal in foreign call params with the current brillig spec